Author: Federico Pellegrin <fede@evolware.org>
Date:   Tue Apr 18 05:21:08 2023 +0200

versioneer: fixes for Python 3.12 compatibility of SafeConfigParser

See for details: https://github.com/python/cpython/pull/92503/files

Upstream: https://github.com/ParallelSSH/ssh-python/pull/71

diff -ruN a/versioneer.py b/versioneer.py
--- a/versioneer.py	2022-12-03 21:54:08.000000000 +0100
+++ b/versioneer.py	2023-04-28 05:58:29.879551262 +0200
@@ -339,9 +339,9 @@
     # configparser.NoOptionError (if it lacks "VCS="). See the docstring at
     # the top of versioneer.py for instructions on writing your setup.cfg .
     setup_cfg = os.path.join(root, "setup.cfg")
-    parser = configparser.SafeConfigParser()
+    parser = configparser.ConfigParser()
     with open(setup_cfg, "r") as f:
-        parser.readfp(f)
+        parser.read_file(f)
     VCS = parser.get("versioneer", "VCS")  # mandatory
 
     def get(parser, name):
